dirty_cat/minhash_encoder.py

```diff
@@ -184,14 +178,53 @@ def transform(self, X):
             X = X[:, 0]
         # Check if first item has str or np.str_ type
         assert isinstance(X[0], str), "ERROR: Input data is not string."
+        if self.minmax_hash:
+            assert self.n_components % 2 == 0,\
+                    "n_components should be even when minmax_hash=True"
+        if self.hashing == 'murmur':
+            assert not(self.minmax_hash),\
+                   "minmax_hash not implemented with murmur"
+        if self.handle_missing not in ['error', '']:
+            template = ("handle_missing should be either 'error' or "
+                        "'', got %s")
+            raise ValueError(template % self.handle_missing)
         X_out = np.zeros((len(X), self.n_components))
+        X = X.reshape(-1)
 
         # TODO Parallel run here
-        for i, x in enumerate(X):
-            if x not in self.hash_dict:
-                self.hash_dict[x] = self.get_hash(x)
+        nan_idx = []
+
+        if self.hashing == 'fast':
+            for i, x in enumerate(X):
+                if isinstance(x, float):
+                    nan_idx.append(i)
+                elif x not in self.hash_dict:
+                    self.hash_dict[x] = self.get_fast_hash(x)
+                    X_out[i, :] = self.hash_dict[x]
```

Comment:
I think that the following would work:
```suggestion
                    X_out[i, :] = self.hash_dict[x] = self.get_fast_hash(x)
```
It should be faster, as it would avoid a value lookup in X_out, which is probably not completely cheap.

Reply:
👍